src/web/mjs/connectors/SmackJeeves.mjs

```diff
@@ -1,130 +1,99 @@
+                        title: this.he.decode( element.title ).trim()
+                    };
+                }));
+            }
+        }
+
+        return manga_list;
     }
 
-    /**
-     *
-     */
-    _getChapterList( manga, callback ) {
-        let comicURL;
-        this.fetchDOM( this.url + manga.id, 'div.middle div.title div.buttons a.button:last-of-type' )
-            .then( data => {
-                comicURL = data[0].href;
-                this.requestOptions.headers.set( 'x-cookie', '_sj_view_m=' + Math.floor( Date.now() / 1000 ) );
-                return this.fetchDOM( comicURL, 'select.jumpbox' );
-            } )
-            .then( data => {
-                if( !data.length ) {
-                    throw new Error( 'Page template not supported, no chapter/page selection box found!' );
-                }
-                let chapterList = [...data[0].querySelectorAll( '.jumpbox_chapter' )];
-                if( !chapterList.length ) {
-                    chapterList = [
-                        {
-                            id: comicURL,
-                            title: manga.title,
-                            language: 'en'
-                        }
-                    ];
-                } else {
-                    chapterList = chapterList.map( element => {
-                        return {
-                            id:  new URL( element.value || comicURL, comicURL ).href,
-                            title: element.label.trim() || element.textContent.trim(),
-                            language: 'en'
-                        };
-                    } );
+    async _getChapters(manga) {
+        const url = new URL('/api'+manga.id, this.url);
+        const data = await this._fetchPOST(url);
 
-                }
-                callback( null, chapterList );
-            } )
-            .catch( error => {
-                console.error( error, manga );
-                callback( error, undefined );
-            } );
+        return data.result.list.map(element => {
+            return {
+                id: element.articleUrl,
+                title: this.he.decode( element.articleTitle ).trim()
+            };
+        });
     }
 
-    /**
-     *
-     */
-    _getPageList( manga, chapter, callback ) {
-        let request = new Request( chapter.id, this.requestOptions );
-        this.fetchDOM( request, 'select.jumpbox option.jumpbox_page' )
-            .then( data => {
-                let pageLinks = data.map( element => this.createConnectorURI( this.getAbsolutePath( element.value, request.url ) ) );
-                callback( null, pageLinks );
-            } )
-            .catch( error => {
-                console.error( error, chapter );
-                callback( error, undefined );
-            } );
+    async _getPages(chapter) {
+        const script = `
+            new Promise((resolve, reject) => {
```

Comment:
Can be simplified to `new Promise(resolve => resolve(cmnDate.comicData));`
A promise will reject exceptions automatically (as long as they are not within an async scope, e.g. setTimeout)